frontend/test/test-utils.tsx

```diff
@@ -180,6 +181,24 @@ export const createMockRouter = (overrides?: Partial<InjectedRouter>) => {
   };
 };
 
+export const createMockLocation = (
+  overrides?: Partial<IRouterLocation>
+): IRouterLocation => {
+  // Default values for the location object
+  const defaultLocation: IRouterLocation = {
+    pathname: "/",
+    host: "localhost:8080",
+    hostname: "localhost",
+    port: "8080",
+    protocol: "http:",
+  };
+
+  return {
+    ...defaultLocation,
+    ...overrides,
+  };
+};
+
```

Comment:
Did you end up using this or IRouterLocation?

Reply:
Did not, but figured it might be useful to leave there for the future